File: ui/src/types/crypto.ts
import type { LogEntry } from './shared'

export interface CryptoSignal {
  market_id: string
  label: string
  side: string
  edge_pct: number
  kelly_size: number
  strategy_id: string
  status: string
  created_at: string
}

export interface CryptoStrategy {
  id: string
  name: string
  description: string
  enabled: boolean
  auto_execute: boolean
}

export interface GammaMarket {
  id: string
  title: string
  best_bid: number
  best_ask: number
  volume: number
}

export interface CryptoState {
  btc_usdt: string
  events: string[]
  strategies: CryptoStrategy[]
  signals: CryptoSignal[]
  markets: GammaMarket[]
  binance_lag_confidence: number[]
  logical_arb_confidence: number[]
  logs: LogEntry[]
}

File: src/commands/dto/crypto.rs
//! Crypto tab DTOs.

use serde::Serialize;

#[derive(Serialize, Clone)]
pub struct CryptoSignalDto {
    pub market_id: String,
    pub label: String,
    pub side: String,
    pub edge_pct: f64,
    pub kelly_size: f64,
    pub strategy_id: String,
    pub status: String,
    pub created_at: String,
}

#[derive(Serialize, Clone)]
pub struct CryptoStrategyDto {
    pub id: String,
    pub name: String,
    pub description: String,
    pub enabled: bool,
    pub auto_execute: bool,
}

#[derive(Serialize, Clone)]
pub struct GammaMarketDto {
    pub id: String,
    pub title: String,
    pub best_bid: f64,
    pub best_ask: f64,
    pub volume: f64,
}

#[derive(Serialize, Clone)]
pub struct CryptoStateDto {
    pub btc_usdt: String,
    pub events: Vec<String>,
    pub strategies: Vec<CryptoStrategyDto>,
    pub signals: Vec<CryptoSignalDto>,
    pub markets: Vec<GammaMarketDto>,
    pub binance_lag_confidence: Vec<f64>,
    pub logical_arb_confidence: Vec<f64>,
    pub logs: Vec<super::shared::LogEntryDto>,
}
